docs/book/src/development/development.md

```diff
@@ -50,9 +50,80 @@ make docker-build docker-push
+# [Optional] Control Plane Machine Flavor (Default: m1.medium)
+export OPENSTACK_CONTROL_PLANE_MACHINE_FLAVOR="<openstack_control_plane_machine_flavor>"
+# [Optional] Node Machine Flavor (Default: m1.small)
+export OPENSTACK_NODE_MACHINE_FLAVOR="<openstack_node_machine_flavor>"
+# [Optional] OpenStack Cloud Environment (Default: capo-e2e)
+export OPENSTACK_CLOUD="<openstack_cloud>"
+```
+
+### Creating a Kind Cluster
+
+Create a Kind cluster. This process involves the initialization of Cluster API providers for OpenStack.
+
+```bash
+kind create cluster
+export CLUSTER_TOPOLOGY=true
+clusterctl init --infrastructure openstack
+```
+
+### Secret Configuration
+
+CAPO needs a clouds.yaml file in order to manage the OpenStack resources needed for the Cluster. This should be supplied as a secret named `${CLUSTER_NAME}-clouds-yaml`. You can create this secret for example with:
+
+```bash
+kubectl create secret generic ${CLUSTER_NAME}-clouds-yaml --from-file=clouds.yaml
+```
+
+### Applying ClusterClass and Cluster Configuration
+
+Use `clusterctl` to substitute variables in your ClusterClass configuration file and apply it to set up the cluster class and cluster.
+
+```bash
+clusterctl generate cluster ${CLUSTER_NAME} --from templates/clusterclass-dev-test.yaml  | kubectl apply -f -
+clusterctl generate cluster ${CLUSTER_NAME} --from templates/cluster-template-development.yaml  | kubectl apply -f -
+```
+
 ## Developing with Tilt
 
-We have support for using [Tilt](https://tilt.dev/) for rapid iterative development. Please visit the [Cluster API documentation on Tilt](https://cluster-api.sigs.k8s.io/developer/tilt.html) for information on how to set up your development environment. 
+We have support for using [Tilt](https://tilt.dev/) for rapid iterative development. Please visit the [Cluster API documentation on Tilt](https://cluster-api.sigs.k8s.io/developer/tilt.html) for information on how to set up your development environment.
+
+To utilize Tilt with ClusterClass, update your `tilt-settings.yaml` file as follows:
+
+```yaml
+kustomize_substitutions:
+  CLUSTER_TOPOLOGY: "true"
+  # [Optional] SSH Keypair Name for Instances in OpenStack (Default: "")
+  OPENSTACK_SSH_KEY_NAME: "<openstack_keypair_name>"
+  # [Optional] Control Plane Machine Flavor (Default: m1.medium)
+  OPENSTACK_CONTROL_PLANE_MACHINE_FLAVOR: "<openstack_control_plane_machine_flavor>"
+  # [Optional] Node Machine Flavor (Default: m1.small)
+  OPENSTACK_NODE_MACHINE_FLAVOR: "<openstack_node_machine_flavor>"
+  # [Optional] OpenStack Cloud Environment (Default: capo-e2e)
+  OPENSTACK_CLOUD: "<openstack_cloud>"
+
+additional_kustomizations:
+  secret_kustomization: /path/to/kustomize/secret/configuration
+```
+
+Ensure the specified path (`/path/to/kustomize/secret/configuration`) contains both the `clouds.yaml` file and a `kustomization.yaml` file. The `kustomization.yaml` should define the necessary resources, such as a Kubernetes secret, using the `clouds.yaml` file.
```

Comment:
as a noob I'll love to find out how you did :) but we can add it later.

Reply:
in fact, we can simply create the secret beforehand

Reply:
> as a noob I'll love to find out how you did :) but we can add it later.
:) What do you mean exactly with later? Do you suggest taking the explanation away for now?

Reply:
if you're fine adding it in this PR, otherwise we can do it later via another PR, that's what I meant.
Thanks

Reply:
> in fact, we can simply create the secret beforehand

Yes you can freely choose how to do it. The additional kustomization is just a convenience to make tilt apply it automatically. :slightly_smiling_face: 

It would probably be a good idea to give an example here. Perhaps something like this:

````suggestion
Ensure the specified path (`/path/to/kustomize/secret/configuration`) contains both the `clouds.yaml` file and a `kustomization.yaml` file. The `kustomization.yaml` should define the necessary resources, such as a Kubernetes secret, using the `clouds.yaml` file.

For example, if you want to automatically create a secret named `dev-test-clouds-yaml` with the content of your `clouds.yaml` every time you do `tilt up`, you could do the following.
Create a folder to hold the kustomization.
We will use `/tmp/capo-dev` as example here.
Add the `clouds.yaml` file that you want to use to the folder.
It could look something like this:

```yaml
clouds:
  capo-e2e:
    auth:
      username: demo
      password: secretadmin
      # If using application credentials you would have something like this instead:
      # auth_type: v3applicationcredential
      # application_credential_id: abc123
      # application_credential_secret: 456def
      user_domain_id: default
      auth_url: https://example.com/identity
      domain_id: default
      project_name: demo
    verify: false
    region_name: RegionOne
```

Create a kustomization file named `kustomization.yaml` in the same folder:

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
# Do not add random hash to the end of the secret name
generatorOptions:
  disableNameSuffixHash: true

secretGenerator:
- files:
  - clouds.yaml
  name: dev-test-clouds-yaml
  type: Opaque
```

If you now add `/tmp/capo-dev` to the `additional_kustomizations`, tilt will automatically apply the secret when you do `tilt up`.
To check that the kustomization produces the desired output, do `kustomize build /tmp/capo-dev`.
````